src/tui/proc-table.go

```diff
@@ -209,6 +221,51 @@ func (pv *pcView) setTableSorter(sortBy ColumnID) {
 	}
 }
 
+func (pv *pcView) getProcessSearchRegex() *regexp.Regexp {
+	pv.processRegexMtx.Lock()
+	defer pv.processRegexMtx.Unlock()
+
+	return pv.processRegex
+}
+
+func (pv *pcView) resetProcessSearch() {
+	pv.processRegexMtx.Lock()
+	defer pv.processRegexMtx.Unlock()
+
+	pv.processRegex = nil
+	go pv.appView.QueueUpdateDraw(func() {
+		pv.fillTableData()
+	})
+}
+
+func (pv *pcView) searchProcess(search string, isRegex, caseSensitive bool) error {
+	pv.processRegexMtx.Lock()
```

Comment:
That's a good start that you added a mutex to the regex assignment, but the use of the `pv.processRegex` is still unprotected during match.
What you'll probably want to do is implement 2 funcs:

```golang
func (pv *pcView) setProcRegex(reg *regexp.Regexp) {
	pv.processRegexMtx.Lock()
	defer pv.processRegexMtx.Unlock()
	//it can be nil when you want to reset
	pv.processRegex = reg
}

func (pv *pcView) matchProcRegex(procName string) bool {
	pv.processRegexMtx.Lock()
	defer pv.processRegexMtx.Unlock()
	if pv.processRegex != nil {
		return pv.processRegex.MatchString(procName)
	}
	return false
}
```

Reply:
That makes sense, i do wonder. Does this need a mutex? 
The `proc-filter` form creates a new regex for every input, and saves the pointer to `pv.processRegex`.
When the `fillTableData` gets the current pointer to the current regex. If another search input it done during the `fillTableData` call, a new regex is created and written to `pv`. `fillTableData` still points to the old regex though, and this regex is not garbage collected. So at worst its slightly outdated?

Reply:
2 different threads writing to the same memory location, even for a new value assignment, is a recipe for disaster :)